### ISSUE TITLE
Ruby mode rbenv error

### ISSUE THREAD

**Issue by Eilie** [syl20bnr/spacemacs#4423]
#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.1
- Spacemacs branch: master (rev. 0f9743a)
- Distribution: spacemacs
- Layers:
```elisp
(git ruby-on-rails html javascript emacs-lisp syntax-checking ruby)
```

#### Backtrace
```
Debugger entered--Lisp error: (void-function rbenv--locate-file)
  (rbenv--locate-file ".ruby-version")
  (let ((version-file-path (rbenv--locate-file ".ruby-version"))) (global-rbenv-mode) (if version-file-path (progn (rbenv-use (rbenv--read-version-from-file version-file-path)) (message (concat "[rbenv] Using ruby version " "from .ruby-version file."))) (message "[rbenv] Using the currently activated ruby.")))
  spacemacs//enable-rbenv()
  run-hooks(change-major-mode-after-body-hook prog-mode-hook ruby-mode-hook)
  apply(run-hooks (change-major-mode-after-body-hook prog-mode-hook ruby-mode-hook))
  run-mode-hooks(ruby-mode-hook)
  ruby-mode()
...
```

Removing following from my `.spacemacs` makes error go away.
```
  ;; User initialization goes here
  (setq-default ruby-version-manager 'rbenv)
```

**Comment by TheBB** [comment]
Confirmed on gitter. `rbenv` must be loaded sooner in the `spacemacs//enable-rbenv` function. Probably enough just to require it?

**Comment by syl20bnr** [comment]
Should be fixed in 0.105.4

**Comment by Eilie** [comment]
@TheBB @syl20bnr It's fixed indeed, thank you!
